File: Python/dayconlientuckhacbiet.py
def solve(arr: list):
    n = len(arr)
    max_len, m = 0, 0
    pos = {}
    for i in range(n):
        if arr[i] in pos:
            m = max(m, pos[arr[i]] + 1)
        pos[arr[i]] = i
        max_len = max(max_len, i - m + 1)
    return max_len
input()
arr = list(map(int, input().split()))
print(solve(arr))

File: Cpp/dayconlientuckhacbiet.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

int solve(const vector<int> &arr, int n) {
	unordered_map<int, int> distinct_elements;
	int result = 0;
	int start = 0;
	for (int end = 0; end < n; end++) {
		if (distinct_elements.find(arr[end]) != distinct_elements.end()) {
			start = max(start, distinct_elements[arr[end]] + 1);
		}
		distinct_elements[arr[end]] = end;
		result = max(result, end - start + 1);
	}

	return result;
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
	int n;
	cin >> n;
	vector<int> arr(n);
	for (int i = 0; i < n; ++i) {
		cin >> arr[i];
	}
	cout << solve(arr, n);
	return 0;
}


// code by duck it1
